sdk/iothub/src/service/twin.rs

```diff
@@ -0,0 +1,419 @@
+    ///     let iothub = ServiceClient::from_connection_string(connection_string, 3600).expect("Failed to create the ServiceClient!");
+    ///     let twin = iothub.update_device_twin("some-device")
+    ///                  .properties(json!({
+    ///                    "PropertyName": "PropertyValue",
+    ///                    "ParentProperty": {
+    ///                      "ChildProperty": "ChildValue"
+    ///                    }
+    ///                  }));
+    /// }
+    pub fn properties(mut self, desired_properties: serde_json::Value) -> Self {
+        self.desired_properties = Some(desired_properties);
+        self
+    }
+
+    /// Set the ETag for the twin
+    ///
+    /// ```
+    /// # let connection_string = "HostName=cool-iot-hub.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=YSB2ZXJ5IHNlY3VyZSBrZXkgaXMgaW1wb3J0YW50Cg==";
+    /// use iothub::service::ServiceClient;
+    ///
+    /// let iothub = ServiceClient::from_connection_string(connection_string, 3600).expect("Failed to create the ServiceClient!");
+    /// let twin = iothub.update_device_twin("some-device")
+    ///                  .if_match("AAAAAAAAAAA=");
+    /// ```
+    pub fn if_match<T>(mut self, if_match: T) -> Self
+    where
+        T: Into<String>,
+    {
+        self.if_match = Some(if_match.into());
+        self
+    }
+
+    /// Updates the twin with the desired settings
+    ///
+    /// ```
+    /// use iothub::service::ServiceClient;
+    /// # #[macro_use]
+    /// # extern crate serde_json;
+    ///
+    /// fn main() {
+    ///     # let connection_string = "HostName=cool-iot-hub.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=YSB2ZXJ5IHNlY3VyZSBrZXkgaXMgaW1wb3J0YW50Cg==";
+    ///     let iothub = ServiceClient::from_connection_string(connection_string, 3600).expect("Failed to create the ServiceClient!");
+    ///     let twin = iothub.update_device_twin("some-device")
+    ///                  .tag("TagName", "TagValue")
+    ///                  .properties(json!({"PropertyName": "PropertyValue"}))
+    ///                  .execute();
+    /// }
+    /// ```
+    pub async fn execute(self) -> Result<R, AzureError> {
+        let desired_twin = DesiredTwin {
+            contents: json!({
+                "properties": {
+                    "desired": self.desired_properties.unwrap_or_else(|| json!({}))
+                },
+                "tags": self.desired_tags
+            }),
+        };
+
+        update_twin(
+            self.service_client,
+            self.method,
+            self.device_id,
+            self.module_id,
+            self.if_match,
+            desired_twin,
+        )
+        .await
+    }
+}
+
+/// Helper function for getting a device/module twin
+async fn get_twin<T>(service_client: &ServiceClient, uri: String) -> Result<T, AzureError>
+where
+    for<'de> T: Deserialize<'de>,
```

Comment:
I think for all of these generic bounds, you want `DeserializedOwned` (https://docs.serde.rs/serde/de/trait.DeserializeOwned.html)